### PR TITLE
Refactors update community

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9051

## Description of Changes
- Refactors old endpoint
- Fixes tests

 
## Test Plan
- Extensive testing of all scenarios to update a community
- Consider namespaces

### CODE REVIEW

File: libs/model/src/community/UpdateCommunity.command.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            attributes: ['url', 'eth_chain_id', 'cosmos_chain_id'],
+          },
+        ],
+      });
+      mustExist('Community', community); // if authorized as admin, community is always found
+
+      // Only permit site admins to update custom domain field on communities, as it requires
+      // external configuration (via heroku + whitelists).
+      // Currently does not permit unsetting the custom domain; must be done manually.
+      if (custom_domain && community.custom_domain !== custom_domain) {
```
Comment (timolegros) [inline libs/model/src/community/UpdateCommunity.command.ts:75]: Shouldn't this be entirely disabled i.e. `custom_domain` should not be in the payload? @kurtisassad

File: libs/model/src/community/UpdateCommunity.command.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      custom_stages && (community.custom_stages = custom_stages);
+
+      await community.save();
+
+      // Suggested solution for serializing BigInts
+      // https://github.com/GoogleChromeLabs/jsbi/issues/30#issuecomment-1006086291
+      (BigInt.prototype as any).toJSON = function () {
+        return this.toString();
+      };
+      return (await models.Community.findOne({ where: { id } }))!.toJSON();
```
Comment (timolegros) [inline libs/model/src/community/UpdateCommunity.command.ts:157]: Can we not just do `return community.toJSON()`?

**Review by timolegros — DISMISSED** [review]
Getting a Zod schema error when I just load the dashboard (sign-in):
```
[08:33:10.550] ERROR (utils.ts): INTERNAL_SERVER_ERROR: [ZodError] community.getCommunities: [
  {
    "validation": "url",
    "code": "invalid_string",
    "message": "Invalid url",
    "path": [
      "results",
      0,
      "terms"
    ]
  },
  {
    "validation": "url",
    "code": "invalid_string",
    "message": "Invalid url",
    "path": [
      "results",
      2,
      "terms"
    ]
  }
] 
```

Will test once this issue + comments are addressed.

**Comment by Rotorsoft** [comment]
> Getting a Zod schema error when I just load the dashboard (sign-in):
> 
> ```
> [08:33:10.550] ERROR (utils.ts): INTERNAL_SERVER_ERROR: [ZodError] community.getCommunities: [
>   {
>     "validation": "url",
>     "code": "invalid_string",
>     "message": "Invalid url",
>     "path": [
>       "results",
>       0,
>       "terms"
>     ]
>   },
>   {
>     "validation": "url",
>     "code": "invalid_string",
>     "message": "Invalid url",
>     "path": [
>       "results",
>       2,
>       "terms"
>     ]
>   }
> ] 
> ```
> 
> Will test once this issue + comments are addressed.

sorry, this should be draft... I'm still making changes before I fully test this locally

**Comment by rbennettcw** [comment]
got a conflict– I want to make sure I test with the final code

**Review by rbennettcw — APPROVED** [review]
seems to work fine for community updates, though I didn't get to test namespaces since there's a wallet bug

File: libs/model/src/community/UpdateCommunity.command.ts (hunk 2 above)
Comment (timolegros) [reply libs/model/src/community/UpdateCommunity.command.ts]: Honestly never seen this before but seems like something that should be set in top-level entry file so we have consistent parsing of BigInts. That said, I don't think we explicitly use native JS BigInts anywhere.

File: libs/model/src/community/UpdateCommunity.command.ts (hunk 1 above)
Comment (kurtisassad) [reply libs/model/src/community/UpdateCommunity.command.ts:75]: Yeah we can remove this because the custom domain stuff is in a different route now.

Comment (timolegros) [reply libs/model/src/community/UpdateCommunity.command.ts:75]: Confirmed with Kurtis that this should be removed.

Comment (Rotorsoft) [reply libs/model/src/community/UpdateCommunity.command.ts]: Removed 👍

File: libs/model/src/community/UpdateCommunity.command.ts (hunk 2 above)
Comment (Rotorsoft) [reply libs/model/src/community/UpdateCommunity.command.ts]: Removed 👍

**Review by kurtisassad — APPROVED** [review]

**Pull request merged**